docs/overview/Lifecycle.md

```diff
@@ -0,0 +1,335 @@
+  const config = getState().getIn(['config', 'my-url']);
+  // we can preload Holocron modules
+  await dispatch(composeModules([{ name: 'my-other-module', props: ownProps }]));
+  // update the store based on API
+  dispatch(
+    myAction(
+      await (await fetchClient(config.url)).json()
+    )
+  );
+}
+
+MyModule.holocron = {
+  name: 'my-module',
+  loadModuleData,
+};
+```
+
+There's a few `Module.appConfig`s that we can utilize to cater to our use cases:
+
+> `Module.appConfig.provideStateConfig` sets the configuration for modules on the server and client
+
+> `Module.appConfig.createSsrFetch` can be used to compose a `fetchClient` that is provided to
+> the `loadModuleData` function
+
+> `Module.appConfig.extendSafeRequestRestrictedAttributes` allows us to include cookies and headers
+> from the request, this will allow `loadModuleData` calls to have credentials if calling an API
+
+Do note, the `ownProps` will vary between server and browser. The browser will receive
+the expected props, however on the server side, the `route` prop will contain the `props`:
+
+```jsx
+async function loadModuleData({ ownProps }) {
+  const { route: props } = ownProps;
+}
+```
+
+**Rendering**
+
+After loading the module data, the root module and the child modules that matched
+the request URL are composed and rendered according to their `path`. This will
+generate the HTML body that will be sent to the server.
+
+If we want to update the `<head />` of the HTML document,
+[`react-helmet`](react-helmet) is bundled with One App and integrated into the
+server side rendering. We can set up our title, meta tags, link styles to the
+document using `Helmet`:
+
+```jsx
+import React from 'react';
+import Helmet from 'react-helmet';
+
+export default function MyModule({ children }) {
+  return (
+    <React.Fragment>
+      <Helmet>
+        <link rel="stylesheet" href="..." />
+        <script src="..." />
+      </Helmet>
+      {children}
+    </React.Fragment>
+  );
+}
+```
+
+**Response**
+
+These are some of the headers we can expect from the
+server-side rendered HTML document:
+
+> * `Content-Security-Policy`: set from `Module.appConfig.csp`
+> * `Cache-Control` & `Pragma`: preset caching by One App
+> * `One-App-Version`: current One App version used
+> * [`helmet`](helmet): security headers provided by `helmet`
+
+### Partials Renders
+
+One App can be configured to statically render Holocron modules as an independent
+entry point when rendered as a partial, regardless of URL path structures
+defined with `Module.childRoutes` and `ModuleRoute` - this is made possible by
+the micro-frontend driven architecture baked into One App, and can be utilized
+to render modules individually. Partial renders gives us the ability to generate
+emails, templates or any static HTML rendered by a Holocron module.
+
+Both the standard and partial rendering follow the same lifecycle,
+with the key difference being the designated entry module being rendered.
```

Comment:
Maybe we should adjust this paragraph, the only difference in partials is dispatching the action to change the rendering method `setPartialOnly` that will set the render method to `https://reactjs.org/docs/react-dom-server.html#rendertostaticmarkup` the rest is the same and the difference mentioned here is managed by the user i.e. "rendering in isolation" is not a feature of One App as such but an implementation detail by the module developers

Reply:
Agree with what you are saying, will update.

Reply:
I have updated (04c4301) and removed the implementation specific "rendering isolation" from describing partial rendering.